go/libraries/doltcore/sqle/dfunctions/dolt_merge_base.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+)
+
+const DoltMergeBaseFuncName = "dolt_merge_base"
+
+type MergeBase struct {
+	expression.BinaryExpression
+}
+
+func NewMergeBase(left, right sql.Expression) sql.Expression {
+	return &MergeBase{expression.BinaryExpression{Left: left, Right: right}}
+}
+
+func (d MergeBase) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	if _, ok := d.Left.Type().(sql.StringType); !ok {
+		return nil, sql.ErrInvalidType.New(d.Left.Type())
+	}
+	if _, ok := d.Right.Type().(sql.StringType); !ok {
+		return nil, sql.ErrInvalidType.New(d.Right.Type())
+	}
+
+	leftSpec, err := d.Left.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+	rightSpec, err := d.Right.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	left, right, err := resolveRefSpecs(ctx, leftSpec.(string), rightSpec.(string))
```

Comment:
Regardless of whether you implement the above or not, you should check to see if either param is null here and return null if so